### PR TITLE
ntp: add capability and s6-init support

### PR DESCRIPTION
**Pull request by Villemoes**

### CODE REVIEW

File: recipes/ntp/ntp.inc [hunk 1]
```diff
@@ -98,3 +106,10 @@ FILES_${PN}-ntpdate = "${sysconfdir}/network/if-up.d/ntpdate"
 # ntp originally includes tickadj. It's split off for inclusion in
 # small firmware images on platforms with wonky clocks (e.g. OpenSlug)
 RDEPENDS_${PN}-ntpd = "${PN}-tickadj ${PN}-config ${PN}-init"
```
Comment (Villemoes) [inline recipes/ntp/ntp.inc]: I can add a patch splitting that off.

File: recipes/ntp/ntp.inc [hunk 2]
```diff
@@ -98,3 +106,10 @@ FILES_${PN}-ntpdate = "${sysconfdir}/network/if-up.d/ntpdate"
...
+
+inherit s6rc
+S6RC_LONGRUN_SERVICES = "ntpd ntpd-log"
+SRC_URI += "file://ntpd.run file://ntpd-log.run"
+RECIPE_FLAGS += "ntpd_s6rc_dependencies"
+DEFAULT_USE_ntpd_s6rc_dependencies = "ntpd-log net"
```
Comment (Villemoes) [inline recipes/ntp/ntp.inc]: Hm, not sure about that one. It's up to the distro to make sure there's something that provides a "net" service, ntpd shouldn't force that to be the netbase recipe. Alternatively the distro/BSP may set ntpd_s6rc_dependencies to something that makes sense for that distro/BSP.

Comment (esben) [inline recipes/ntp/ntp.inc]: Fair enough.

File: recipes/ntp/ntp.inc (hunk 1 above)
Comment (esben) [inline recipes/ntp/ntp.inc]: That would be nice.  Will you add it to this PR, or make a new one?

Comment (Villemoes) [inline recipes/ntp/ntp.inc]: Just added it, it does kinda belong with the one adding support for an alternative init.

**Pull request merged**